db/internal_stats.cc

```diff
@@ -982,6 +988,35 @@ void InternalStats::DumpCFMapStats(
   (*levels_stats)[-1] = sum_stats;  //  -1 is for the Sum level
 }
 
+void InternalStats::DumpCFMapStatsIOStalls(
+        std::map<std::string, std::string>* cf_stats) {
+  (*cf_stats)["io_stalls.level0_slowdown"] =
+          std::to_string(cf_stats_count_[LEVEL0_SLOWDOWN_TOTAL]);
+  (*cf_stats)["io_stalls.level0_slowdown_with_compaction"] =
+          std::to_string(cf_stats_count_[LEVEL0_SLOWDOWN_WITH_COMPACTION]);
+  (*cf_stats)["io_stalls.level0_numfiles"] =
+          std::to_string(cf_stats_count_[LEVEL0_NUM_FILES_TOTAL]);
+  (*cf_stats)["io_stalls.level0_numfiles_with_compaction"] =
+          std::to_string(cf_stats_count_[LEVEL0_NUM_FILES_WITH_COMPACTION]);
+  (*cf_stats)["io_stalls.stop_for_pending_compaction_bytes"] =
+          std::to_string(cf_stats_count_[HARD_PENDING_COMPACTION_BYTES_LIMIT]);
+  (*cf_stats)["io_stalls.slowdown_for_pending_compaction_bytes"] =
+          std::to_string(cf_stats_count_[SOFT_PENDING_COMPACTION_BYTES_LIMIT]);
+  (*cf_stats)["io_stalls.memtable_compaction"] =
+          std::to_string(cf_stats_count_[MEMTABLE_COMPACTION]);
+  (*cf_stats)["io_stalls.memtable_slowdown"] =
+          std::to_string(cf_stats_count_[MEMTABLE_SLOWDOWN]);
+
+  uint64_t total =
+          cf_stats_count_[LEVEL0_SLOWDOWN_TOTAL] +
+          cf_stats_count_[LEVEL0_NUM_FILES_TOTAL] +
+          cf_stats_count_[SOFT_PENDING_COMPACTION_BYTES_LIMIT] +
+          cf_stats_count_[HARD_PENDING_COMPACTION_BYTES_LIMIT] +
+          cf_stats_count_[MEMTABLE_COMPACTION] + cf_stats_count_[MEMTABLE_SLOWDOWN];
```

Comment:
There are 2 kinds of stalls
slow down stalls -> Writes continue but with slow down penalty
stopping stalls -> Writes totally stop

I think we should have two different counters ("total_slowdown_stall", "total_stop_stall"